CMakeLists.txt

```
@@ -166,11 +165,17 @@ include_directories(
         ${CMAKE_BINARY_DIR}/include
         ${PCRE_INCLUDE_DIRS}
 )
+
 if (OPENSSL_FOUND)
     include_directories(${OPENSSL_INCLUDE_DIR})
     link_libraries(${OPENSSL_CRYPTO_LIBRARY})
 endif(OPENSSL_FOUND)
 
+if (OPENSSL_ROOT_DIR)
+    get_filename_component(OPENSSL_LIBRARY_DIR ${OPENSSL_CRYPTO_LIBRARY} DIRECTORY)
+    set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_RPATH}:${OPENSSL_LIBRARY_DIR}")
+endif(OPENSSL_ROOT_DIR)
+
```

```suggestion
set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
```

https://cmake.org/cmake/help/v3.26/variable/CMAKE_INSTALL_RPATH_USE_LINK_PATH.html
Would this not work?

---

Yes, that works and simplifies things.  I tested it and works well:

```
08:43:26 zeus:(cmake_boringssl_rpath)~/dev/apache/trafficserver$ objdump -x /opt/ats/bin/traffic_server
...
  RUNPATH              /opt/ats/lib:/opt/boringssl/lib64
```